freqtrade/templates/FreqaiHybridExampleStrategy.py

```diff
@@ -0,0 +1,337 @@
+                )["STX"]
+
+        for multiplier in self.sell_m3.range:
+            for period in self.sell_p3.range:
+                dataframe[f"supertrend_3_sell_{multiplier}_{period}"] = self.supertrend(
+                    dataframe, multiplier, period
+                )["STX"]
+
+        dataframe = self.freqai.start(dataframe, metadata, self)
+
+        return dataframe
+
+    def populate_entry_trend(self, df: DataFrame, metadata: dict) -> DataFrame:
+
+        # User now can use their custom strat creation in addition to their
+        # future prediction "up" or "down".
+
+        df.loc[
+            (df[f"supertrend_1_buy_{self.buy_m1.value}_{self.buy_p1.value}"] == "up") &
+            (df[f"supertrend_2_buy_{self.buy_m2.value}_{self.buy_p2.value}"] == "up") &
+            (df[f"supertrend_3_buy_{self.buy_m3.value}_{self.buy_p3.value}"] == "up") &
+            (df["do_predict"] == 1) &
+            (df['&s-up_or_down'] == 'up'),
+            "enter_long",
+        ] = 1
+
+        df.loc[
+            (df[f"supertrend_1_sell_{self.sell_m1.value}_{self.sell_p1.value}"] == "down") &
+            (df[f"supertrend_2_sell_{self.sell_m2.value}_{self.sell_p2.value}"] == "down") &
+            (df[f"supertrend_3_sell_{self.sell_m3.value}_{self.sell_p3.value}"] == "down") &
+            (df["do_predict"] == 1) &
+            (df['&s-up_or_down'] == 'down'),
+            "enter_short",
+        ] = 1
+
+        return df
+
+    def populate_exit_trend(self, df: DataFrame, metadata: dict) -> DataFrame:
+
+        df.loc[
+            (df[f"supertrend_2_sell_{self.sell_m2.value}_{self.sell_p2.value}"] == "down"),
+            "exit_long",
+        ] = 1
+
+        df.loc[
+            (df[f"supertrend_2_buy_{self.buy_m2.value}_{self.buy_p2.value}"] == "up"),
+            "exit_short",
+        ] = 1
+
+        return df
+
+    def get_ticker_indicator(self):
+        return int(self.config["timeframe"][:-1])
+
+    def confirm_trade_entry(self, pair: str, order_type: str, amount: float,
+                            rate: float, time_in_force: str, current_time, entry_tag, side: str,
+                            **kwargs, ) -> bool:
+
+        df, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
+        last_candle = df.iloc[-1].squeeze()
+
+        if side == "long":
+            if rate > (last_candle["close"] * (1 + 0.0025)):
+                return False
+        else:
+            if rate < (last_candle["close"] * (1 - 0.0025)):
+                return False
+
+        return True
+
+    def leverage(self, pair: str, current_time: datetime, current_rate: float,
+                 proposed_leverage: float, max_leverage: float, entry_tag: Optional[str], side: str,
+                 **kwargs) -> float:
+
+        return 1
+
+    """
+    Supertrend Indicator; adapted for freqtrade, optimized by the math genius.
+    from: Perkmeister#2394
+    """
+
+    def supertrend(self, dataframe: DataFrame, multiplier, period):
```

Comment:
Sample strategies should not implement indicators by copy/paste. That's bad practice for any strategy - so these strategies clearly should be leading by example in implementing best practices by reusing existing indicators.

supertrend is available in multiple libraries, including technical, pandas-ta (both of which freqtrade depends on).

Reply:
Fair point. I guess the reason this was done is because this supertrend function is far more efficient than the standard libs. 

But an example is an example, we will remove the function and use pandas-ta instead.

Reply:
if it is (and results are still identical) - it should be contributed to one of the mentioned libraries.

Reply:
Thanks for checking on that. Resolved it by removing supertrend all together.